tests/Form/DataTransformerResolverTest.php

```diff
@@ -0,0 +1,221 @@
+    {
+        $this->fieldDescription = $this->prophesize(FieldDescriptionInterface::class);
+        $this->modelManager = $this->prophesize(ModelManagerInterface::class);
+        $this->resolver = new DataTransformerResolver();
+    }
+
+    public function testFailedResolve(): void
+    {
+        $this->assertNull($this->resolve());
+    }
+
+    public function provideFieldTypes(): array
+    {
+        return [
+            ['foo'],
+            // override predefined transformers
+            ['date'],
+            ['boolean'],
+            ['choice'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideFieldTypes
+     */
+    public function testResolveCustomDataTransformer(string $fieldType): void
+    {
+        $customDataTransformer = new CallbackTransformer(static function ($value): string {
+            return (string) (int) $value;
+        }, static function ($value): bool {
+            return filter_var($value, FILTER_VALIDATE_BOOLEAN);
+        });
+        $this->fieldDescription->getOption('data_transformer')->willReturn($customDataTransformer);
+        $this->fieldDescription->getType()->willReturn($fieldType);
+
+        $dataTransformer = $this->resolve();
+
+        $this->assertInstanceOf(DataTransformerInterface::class, $dataTransformer);
+        $this->assertSame($customDataTransformer, $dataTransformer);
+    }
+
+    public function getTimeZones(): iterable
+    {
+        $default = new \DateTimeZone(date_default_timezone_get());
+        $custom = new \DateTimeZone('Europe/Rome');
+
+        return [
+            'empty timezone' => [null, $default],
+            'disabled timezone' => [false, $default],
+            'default timezone by name' => [$default->getName(), $default],
+            'default timezone by object' => [$default, $default],
+            'custom timezone by name' => [$custom->getName(), $custom],
+            'custom timezone by object' => [$custom, $custom],
+        ];
+    }
+
+    /**
+     * @dataProvider getTimeZones
+     */
+    public function testResolveDateDataTransformer($timezone, \DateTimeZone $expectedTimezone): void
+    {
+        $this->fieldDescription->getOption('data_transformer')->willReturn(null);
+        $this->fieldDescription->getOption('timezone')->willReturn($timezone);
+        $this->fieldDescription->getType()->willReturn('date');
+
+        $dataTransformer = $this->resolve();
+
+        $this->assertInstanceOf(DateTimeToStringTransformer::class, $dataTransformer);
+
+        $value = '2020-12-12';
+        $defaultTimezone = new \DateTimeZone(date_default_timezone_get());
+        $expectedDate = new \DateTime($value, $expectedTimezone);
+        $expectedDate->setTimezone($defaultTimezone);
+
+        $resultDate = $dataTransformer->reverseTransform($value);
+
+        $this->assertInstanceOf(\DateTime::class, $resultDate);
+        $this->assertSame($expectedDate->format('Y-m-d'), $resultDate->format('Y-m-d'));
```

Comment:
```suggestion
        $this->assertSame($expectedDate->format(\DateTimeInterface::ATOM), $resultDate->format(\DateTimeInterface::ATOM));
```

Reply:
This is unnecessary since we are comparing a date without a time.

Reply:
>we are comparing a date without a time.

That's the point. Using `\DateTimeInterface::ATOM` we can be sure there is no time information in this object.